### DISCUSSION TITLE
Add a queue `full()` API

### DISCUSSION THREAD

**Discussion by brendan-simon-indt** [peter-wangxu/persist-queue#193] [Ideas]
I notice there is a queue `empty()` API but not a queue `full()` API.  Standard python queues have both, so thought it made sense to provide the same for persistent queues.

```python
   def full(self):
        return self.qsize() == self.maxsize
```

I want to remove an item from the front of the queue if it is full before pushing a new item to the back of the queue - and do something with the old item that was removed.  Basically, I'm using the queue is a FIFO of fixed size and what old items to pop off the end.

The `put()` API doesn't have a return value (e.g. a popped off item from a full queue), but that's consistent with `queue.Queue`, so I can't do an atomic `put` and `get` (if full).  That's why the `full()` API would be handy (not that it's hard to explicitly test the queue size (`pq.qsize() == pq.maxsize`)

So would you consider adding the `full()` API or a PR to complement the `empty()` API and make it consistent with `queue.Queue`?

**Comment by peter-wangxu** [comment]
Looks like `full()` api is available since python3.x but not exist in python2.x
I am fine to add this api.
Do you mind sending a new PR for this?

**Comment by brendan-simon-indt** [comment]
Done.  See PR #194
